transcom/mymove
# B-19175-INT SC Customer Search

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19175)

## Summary

Large PR so will try to provide as much detail as possible to help you work through it. This new search allows customers profiles to be searched instead of just moves. Previously, customers could only be accessed via the move, so I added a new `BASE_COUNSELING_CUSTOMER_PATH` and use the customer id. I wrapped the entire customer search tab in a feature flag and moved the create move and add customer buttons into the customer search component. In my testing instructions below, I have included what files are touched along the path of using this component, so hopefully thats helpful.

### How to test

1. Login as a Services Counselor and click Customer Search
-ServicesCounselingQueue - parent component housing the tabs, search form, and results components. Changes here include making the customer search tab conditionally rendered based off the feature flag, adding the search results form/table, removing the add customer button from the customer move search, adding searchType to be handed down to SearchResultsTable (default value of 'moves')
-CustomerSearchForm - handles the radio buttons and input for search (largely copied form MoveSearchForm)
-SearchResultsTable - displays the search results. change columns displayed based on searchType handed down from ServicesCounselingQueue
2. Search for a customer in the customer search. Note the appearance of the Add Customer button after search is clicked
-queries.js - useCustomerSearchQueries (activated by SearchResultsTable, pass down from ServicesCounselingQueue)
-ghcApi.js - searchCustomers (activated by useCustomerSearchQueries)
-handlers/ghcapi/customers.go - passes the search params to customer_searcher.go, receives the search results back, passes that data to model_to_payload.go, then returns that data to the front end
-customer_searcher.go - searches users/service_members based off the entered data
-model_to_payload.go - converts customer data to payload
3. Click Add Customer button to add a customer. Fill out required data, using a unique name, and create.
4. Click move search and search for your customer. Note that you don't find your customer.
5. Click customer search and search for your customer. When they appear in the results, click Create New Move
-routes.js - added routes to access these pages via the customer rather than the move
-src/pages/Office/index.jsx - added `CreateMoveCustomerInfo` and `ServicesCounselingAddOrders`
6. Confirm your customer info and advance to add orders page. Fill out orders info and confirm you're directed to move details page.
7. Return to SC landing page and click move search. Confirm you can now search for your new move
